DataFormats/DetId/interface/DetId.h

```diff
@@ -23,7 +23,7 @@ class DetId {
   static const int kSubdetOffset       = 25;
 
 
-  enum Detector { Tracker=1,Muon=2,Ecal=3,Hcal=4,Calo=5,Forward=6 };
+  enum Detector { Tracker=1,Muon=2,Ecal=3,Hcal=4,Calo=5,Forward=6, Trigger = 7 };
```

Comment:
Trigger DetIds have always been of the type of the detector they represent a trigger for, see examples in CSC, HCAL, etc. Better to make HGCalSubDetId of type "trigger", this what you always know what detector you are talking about.